### DISCUSSION TITLE
Accessing logged in user from a custom lookup script .net5

### DISCUSSION THREAD

**Discussion by woody1987** [serenity-is/Serenity#5578] [Help]
In previous versions of serenity I could use `Authorization.UserDefinition` pretty much anywhere to get the logged in users definition.

In .net5 I can get this using `User.GetUserDefinition(_UserRetrieveService) as UserDefinition;`

Problem is, how do I do this from a custom lookup script `RowLookupScript`?. I can do this in endpoints and repositories easily using `[FromServices] IUserRetrieveService userRetrieveService` and the ClaimsPrincipal `User` is available in those classes. The same cannot be said for elsewhere.

**Comment by erehtolleh** [comment]
did you try to inject it into the constructor of your custom lookupscript?
https://serenity.is/Docs/migration/v3-to-v5
Else where you would do something like below:

```
protected ITwoLevelCache Cache { get; }
protected ISqlConnections SqlConnections { get; }
public ITypeSource TypeSource { get; }
protected IUserAccessor UserAccessor { get; }

public PermissionService(ITwoLevelCache cache, ISqlConnections sqlConnections, 
    ITypeSource typeSource, IUserAccessor userAccessor)
{
    Cache = cache ?? throw new ArgumentNullException(nameof(cache));
    SqlConnections = sqlConnections ?? throw new ArgumentNullException(nameof(sqlConnections));
    TypeSource = typeSource ?? throw new ArgumentNullException(nameof(typeSource));
    UserAccessor = userAccessor ?? throw new ArgumentNullException(nameof(userAccessor));
}
```
